apis/python/src/tiledbsoma/_util.py

```diff
@@ -260,3 +262,26 @@ def ms_to_datetime(millis: int) -> datetime.datetime:
     secs, millis = divmod(millis, 1000)
     dt = datetime.datetime.fromtimestamp(secs, tz=datetime.timezone.utc)
     return dt.replace(microsecond=millis * 1000)
+
+
+def to_clib_result_order(result_order: options.ResultOrderStr) -> clib.ResultOrder:
+    to_clib_result_order = {
+        options.ResultOrder.AUTO: clib.ResultOrder.automatic,
+        options.ResultOrder.ROW_MAJOR: clib.ResultOrder.rowmajor,
+        options.ResultOrder.COLUMN_MAJOR: clib.ResultOrder.colmajor,
+        "auto": clib.ResultOrder.automatic,
+        "row-major": clib.ResultOrder.rowmajor,
+        "column-major": clib.ResultOrder.colmajor,
```

Comment:
you can avoid this duplication by using the fact that you can turn a string into the enum value:

```python
result_order = options.ResultOrder(result_order)
to_clib_result_order = {...}
try:
  return to_clib_result_order
except KeyError as ke:
  raise ValueError(...) from ke